source/Calamari.Aws/Kubernetes/Discovery/AwsKubernetesDiscoverer.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amazon;
+using Amazon.EKS;
+using Amazon.EKS.Model;
+using Calamari.Common.Features.Discovery;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes.Aws;
+using Newtonsoft.Json;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Aws.Kubernetes.Discovery
+{
+    public class AwsKubernetesDiscoverer : IKubernetesDiscoverer
+    {
+        readonly ILog log;
+
+        public AwsKubernetesDiscoverer(ILog log)
+        {
+            this.log = log;
+        }
+
+        public string Name => "Aws";
+        
+        public IEnumerable<KubernetesCluster> DiscoverClusters(string contextJson, IVariables variables)
+        {
+            if (!TryGetAuthenticationDetails(contextJson, out var authenticationDetails))
+                yield break;
+            var workerPool = variables.Get("Octopus.Aws.WorkerPool");
```

Comment:
We can't rely on using variables inside target discovery.

While this will work at present, there will be a change soon whereby the discovery action no longer be using the same ID as the action which triggered target discovery.

That means that any variables which are scoped to the triggering action will not be available during target discovery, and we should therefore avoid using any variables inside discovery.

The exception is the variable which holds the target discovery scope, which will still be explicitly be scoped to the discovery action.

The worker pool ID will be available on the discovery scope, so if we want to log it here, then that's where it can be fetched from.